### PR TITLE
fix(feature flags): disallow groups on early access flags

### PR DESCRIPTION
**Pull request by jurajmajerik**
## Problem
There's a bug resulting from having groups on feature flags linked to early access features: https://posthog.sentry.io/issues/4870109410/?alert_rule_id=14548021&alert_type=issue&notification_uuid=0c490c3f-b8cb-48ff-b601-ea2fdc654f63&project=1899813&referrer=slack

### Backend:
There's a few ways groups can end up on a flag linked to an early access feature:

1) creating an early access feature, linking a group flag - done previously
2) updating an early access feature, linking a group flag - fixed
3) updating a flag linked to an early access feature - fixed

### Frontend:
- Disable the group options from the _Match by_ select when the flag has an early access feature linked. Swapped original _Select_ to _LemonSelect_ to be able to add the disabled reason.

image

## How did you test this code?
- API test for 3), 👀

### CODE REVIEW

File: posthog/api/feature_flag.py [hunk 1]
```diff
@@ -283,6 +283,15 @@ def update(self, instance: FeatureFlag, validated_data: Dict, *args: Any, **kwar
             raise exceptions.ValidationError(
                 "Cannot delete a feature flag that is in use with early access features. Please delete the early access feature before deleting the flag."
             )
+
+        if "get_filters" in validated_data:
```
Comment (neilkakkar) [inline posthog/api/feature_flag.py:287]: can you move this check to `validate_filters` instead? All similar checks exist there

File: posthog/api/test/test_feature_flag.py [hunk 2]
```diff
@@ -3621,6 +3622,39 @@ def test_creating_static_cohort(self):
...
+            "filters": {
+                "aggregation_group_type_index": 2,
+                "groups": [{"properties": [], "rollout_percentage": 100}],
+            }
+        }
+        response = self.client.patch(
+            f"/api/projects/{self.team.id}/feature_flags/{feature_flag.id}/", update_data, format="json"
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
```
Comment (neilkakkar) [inline posthog/api/test/test_feature_flag.py:3656]: assert validation error as well?

**Review by neilkakkar — COMMENTED** [review]
Nice work! Very happy with the de-antifying work :D

File: posthog/api/early_access_feature.py [hunk 3]
```diff
@@ -84,6 +84,11 @@ def update(self, instance: EarlyAccessFeature, validated_data: Any) -> EarlyAcce
 
             related_feature_flag = instance.feature_flag
             if related_feature_flag:
+                if related_feature_flag.aggregation_group_type_index is not None:
```
[the remark this answers is not in the document]
Comment (jurajmajerik) [reply posthog/api/early_access_feature.py:87]: indeed, but theoretically can be done via the API? (i.e. point the early access feature to a flag with groups)

[the remark this answers is not in the document]
Comment (neilkakkar) [reply posthog/api/early_access_feature.py:87]: how so? Even with the API, I don't see a code path that ever updates `instance.feature_flag` 🤔 

if it existed, then wherever it did, we'd totally want this validation, yes!

[the remark this answers is not in the document]
Comment (jurajmajerik) [reply posthog/api/early_access_feature.py]: got it, removed :)

**Review by neilkakkar — APPROVED** [review]
🚀

**Pull request merged**